### PR TITLE
Workflow job execution doc edited to clarify that 'env' context is not available in job 'if' conditional 

### LINKED ISSUE #21927: Workflow job execution doc should clarify that 'env' context is not available in job 'if' conditional
### Code of Conduct

- [X] I have read and agree to the GitHub Docs project's [Code of Conduct](https://github.com/github/docs/blob/main/CODE_OF_CONDUCT.md)

### What article on docs.github.com is affected?

https://docs.github.com/en/actions/using-jobs/using-conditions-to-control-job-execution

### What part(s) of the article would you like to see updated?

The doc has an example:

```yaml
name: example-workflow
on: [push]
jobs:
  production-deploy:
    if: github.repository == 'octo-org/octo-repo-prod'
```

I spent several hours trying to access `env.MY_ENVAR`, running into errors, before finally realizing that the `env` context cannot be accessed until after the job runner has been instantiated (which happens after the `if` guard has passed).

The doc should be updated to clarify that the `env` context is not available in the `if` conditional of a `job`.

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by OlayinkaAtobiloye**
### Why:

Closes https://github.com/github/docs/issues/21927, https://github.com/github/docs/issues/17099

### What's being changed (if available, include any code snippets, screenshots, or gifs):

Screenshot of app with current change.

### Check off the following:

- [x] I have reviewed my changes in staging (look for the "Automatically generated comment" and click the links in the "Preview" column to view your latest changes).
- [x] For content changes, I have completed the [self-review checklist](https://github.com/github/docs/blob/main/contributing/self-review.md#self-review).

### CODE REVIEW

**Comment by OlayinkaAtobiloye** [comment]
Hi, @cmwilson21! I raised a PR addressing a workflow job execution, could you please check it out when chanced? Thanks!

**Comment by cmwilson21** [comment]
@OlayinkaAtobiloye great work! I'll get this triaged for review! ⚡

**Comment by Esiakha** [comment]
[self-review checklist](https://github.com/github/docs/blob/main/contributing/self-review.md#self-review).

**Comment by Esiakha** [comment]
#23859

**Comment by OlayinkaAtobiloye** [comment]
> [self-review checklist](https://github.com/github/docs/blob/main/contributing/self-review.md#self-review).

Hi @Esiakha, it is now completed!

File: content/actions/using-jobs/using-conditions-to-control-job-execution.md [hunk 1]
```diff
@@ -26,3 +26,9 @@ miniTocMaxHeadingLevel: 4
 You would see the following status on a skipped job:
 
 ![Skipped-required-run-details](/assets/images/help/repository/skipped-required-run-details.png)
+
+{% note %}
+
+**Note:** In some parts of the workflow, you cannot use environment variables; but instead, you can use contexts to access the value of an environment variable. For more information, see ["Using the env context to access environment variable values"](https://docs.github.com/en/actions/learn-github-actions/variables#using-the-env-context-to-access-environment-variable-values).
```
Comment (hubwriter) [inline content/actions/using-jobs/using-conditions-to-control-job-execution.md:32]: Style-guide changes:

```suggestion
**Note:** In some parts of the workflow you cannot use environment variables. Instead you can use contexts to access the value of an environment variable. For more information, see "[Using the env context to access environment variable values](/actions/learn-github-actions/variables#using-the-env-context-to-access-environment-variable-values)."
```

**Review by hubwriter — DISMISSED** [review]
@OlayinkaAtobiloye - Many thanks for making these documentation changes - much appreciated. 🙇 

I've made a few small changes, just to keep things in line with our style guide. 

I'll get this merged/published.

Thanks again for contributing to the docs. 
🚀

**Review by hubwriter — APPROVED** [review]

**Comment by avalosriveraamericacitlali-cell** [comment]
> ### Por qué:
> Cierra los números #41927 y [#170.](https://github.com/github/docs/issues/17099)
> 
> ### ¿Qué se va a cambiar? (Si es posible, incluya fragmentos de código, capturas de pantalla o GIF):
> Captura de pantalla de la aplicación con el cambio actual.
> 
> ### Marque las siguientes casillas:
> * [x] He revisado mis cambios en el entorno de pruebas (busca el comentario "Comentario generado automáticamente" y haz clic en los enlaces de la columna "Vista previa" para ver tus últimos cambios).
> * [x] Para los cambios de contenido, he completado la [lista de verificación de autoevaluación](https://github.com/github/docs/blob/main/contributing/self-review.md#self-review) .

**Pull request merged**